ant/job_miner.go

```diff
@@ -1,57 +1,65 @@
 package ant
 
 import (
-	"fmt"
 	"log"
 	"time"
 
 	"github.com/NebulousLabs/Sia/api"
+	"github.com/NebulousLabs/Sia/types"
 )
 
-// blockMining unlocks the wallet and mines some currency.  If more than 100
-// seconds passes before the wallet has received some amount of currency, this
-// job will print an error.
-func (j *jobRunner) blockMining() {
+// blockMining mines blocks until desiredBalance is reached. If desiredBalance
+// is zero, blockMining will start the miner and return, mining indefinitely.
+func (j *jobRunner) blockMining(desiredBalance types.Currency) {
 	j.tg.Add()
 	defer j.tg.Done()
 
-	err := j.client.Post("/wallet/unlock", fmt.Sprintf("encryptionpassword=%s&dictionary=%s", j.walletPassword, "english"), nil)
+	minerRunning := true
+	err := j.client.Get("/miner/start", nil)
 	if err != nil {
 		log.Printf("[%v blockMining ERROR]: %v\n", j.siaDirectory, err)
 		return
 	}
 
-	err = j.client.Get("/miner/start", nil)
-	if err != nil {
-		log.Printf("[%v blockMining ERROR]: %v\n", j.siaDirectory, err)
+	// If desiredBalance is zero, just return, leaving the miner running.
+	runForever := desiredBalance.Cmp(types.ZeroCurrency) == 0
```

Comment:
correct. by not including miner in the `Jobs`, the miner never gets run, regardless of `DesiredCurrency`.